### DISCUSSION TITLE
VisionEye Mapping - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#7159] [Docs]
# VisionEye Mapping - Ultralytics YOLOv8 Docs

VisionEye View Object Mapping using Ultralytics YOLOv8

https://docs.ultralytics.com/guides/vision-eye/

**Comment by RizwanMunawar** [comment]
VisionEye Objects Mapping using Ultralytics YOLOv8 🔥🚀

**Comment by GIGAParviz** [comment]
track_ids = result[0].boxes.id.int().cpu().tolist()
 

AttributeError: 'NoneType' object has no attribute 'int'




i've  got this error how i can fix it?

**Reply by RizwanMunawar** [reply]
@GIGAParviz Thank you for bringing this issue to our attention. You can utilize the provided updated code to address and resolve the reported issue effectively.

```python
import cv2
from ultralytics import YOLO
from ultralytics.utils.plotting import colors, Annotator

model = YOLO("yolov8n.pt")
cap = cv2.VideoCapture("path/to/video/file.mp4")

out = cv2.VideoWriter('visioneye-pinpoint.avi', cv2.VideoWriter_fourcc(*'MJPG'),
                      30, (int(cap.get(3)), int(cap.get(4))))

center_point = (-10, int(cap.get(4)))

while True:
    ret, im0 = cap.read()
    if not ret:
        print("Video frame is empty or video processing has been successfully completed.")
        break

    annotator = Annotator(im0, line_width=2)

    results = model.track(im0, persist=True)
    boxes = results[0].boxes.xyxy.cpu()

    if results[0].boxes.id is not None:
        track_ids = results[0].boxes.id.int().cpu().tolist()

        for box, track_id in zip(boxes, track_ids):
            annotator.box_label(box, label=str(track_id), color=colors(int(track_id)))
            annotator.visioneye(box, center_point)

    out.write(im0)
    cv2.imshow("visioneye-pinpoint", im0)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

out.release()
cap.release()
cv2.destroyAllWindows()
```

Hope this helps! Let me know if you have any additional questions.

Thanks and Regards
Ultralytics Team!

**Reply by GIGAParviz** [reply]
Thanks 💚 that's really cool

**Reply by pderrenger** [reply]
@GIGAParviz you're welcome! 💚 If you have any more questions or need further assistance, feel free to ask. Happy coding with YOLOv8! 🚀

**Comment by leongjy2005** [comment]
Traceback (most recent call last):
  File "d:\Random_YOLO8_Tutorials\Human_Detection\visioneye.py", line 28, in <module>
    annotator.visioneye(box, center_point)
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'Annotator' object has no attribute 'visioneye'

**Reply by leongjy2005** [reply]
I'm facing this issue, PLEASE HELP😥

**Reply by pderrenger** [reply]
@leongjy2005 it seems like you're encountering an `AttributeError` because the `Annotator` object does not have a method named `visioneye`. This could be due to several reasons:

1. The `visioneye` method might not exist in the version of the Ultralytics YOLOv8 library you are using.
2. There could be a typo in the method name.
3. The method might be part of a custom implementation that hasn't been correctly integrated into the `Annotator` class.

To resolve this issue, please ensure the following:
- You have the latest version of the Ultralytics YOLOv8 library installed. You can update the library using pip:
  ```bash
  pip install --upgrade ultralytics
  ```
- Double-check the documentation or source code of the `Annotator` class to confirm the correct method name.
- If `visioneye` is a custom method, ensure it is correctly defined in the `Annotator` class and that the class is properly imported.

If you continue to face issues, consider reaching out on the [Ultralytics YOLOv8 GitHub repository](https://github.com/ultralytics/ultralytics/issues) for more specific guidance. The community and maintainers there can provide more targeted support for your problem. Remember to provide as much context as possible, including the version of the library you're using and any custom code you've added.

**Reply by leongjy2005** [reply]
Alright! Problem Solved, Thanks!

**Reply by glenn-jocher** [reply]
Hey there! 🎉

Glad to hear your issue is resolved! If you have any more questions or need further assistance in the future, don't hesitate to drop by. Happy coding with YOLOv8 and VisionEye! 😊
